pyccel/codegen/printing/cwrappercode.py

```diff
@@ -877,6 +897,20 @@ def _print_FunctionDef(self, expr):
 
         # Call free function for python type
         wrapper_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+        # Call free function for C type
+        if self._target_language == 'c':
+            to_free = ["strides", "shape"]
+            wrapper_body += [Deallocate(
+                                DottedVariable(
+                                            NativeInteger(),
+                                                member,
+                                                lhs=i.name,
+                                                is_pointer=True
+                                            ),
+                                free_function="free",
+                                )
```

Comment:
Doesn't this do the same thing as calling `Deallocate` on an array pointer?

Reply:
yes, but I made some changes to the deallocate class to work for other types.

Reply:
Yes I know. My question is about whether those changes are necessary?

Reply:
If you need to free both the `strides` and `shape` i don't think it's necessary to change the Deallocation node when we discussed this earlier I thought that you need to free the `strides` **alone**.